lib/ansible/plugins/connection/ssh.py

```diff
@@ -279,7 +278,25 @@
           cli:
             - name: private_key_file
               option: '--private-key'
-
+      private_key:
+          description:
+            - Private key contents in PEM format. Requires the C(SSH_AGENT) configuration to be enabled.
+          type: string
+          env:
+            - name: ANSIBLE_PRIVATE_KEY
+          vars:
+            - name: ansible_private_key
```

Comment:
we also probably want `ansible_ssh_private_key` variants, both to be consistent and allow other plugins to use `ansbile_private_key` as a generic, overridable by plugin specific vars.

Reply:
Done.